### PR TITLE
Fix the relationship between the Organization and Token entities

### PR DESCRIPTION
**Pull request by alex-ko-dev**
### Description

Updated the project according to [changes in DataStore API](https://github.com/checkmarx-ltd/Integrations-Datastore/pull/10). 
Refactored access-token related logic and SCM services. SCM services are now aligned with each other (similar logic and naming).

### References

User story [368](https://dev.azure.com/CxFlow/CxFlow/_workitems/edit/368).

### Testing

The tests were fixed according to the changes.
Also tested manually using the frontend app.

### CODE REVIEW

**Comment by cxflowtestuser** [comment]
Scan submitted to Checkmarx

**Comment by cxflowtestuser** [comment]
Logo<br />**Checkmarx SAST - Scan Summary & Details**
### Cx-SAST Summary
Total of 67 vulnerabilities<br />High&nbsp;**0 High**<br />Medium&nbsp;**0 Medium**<br />Low&nbsp;**63 Low**<br />Info&nbsp;**4 Info**<br />
### Violation Summary
#### No policy violation found
***
Logo
**Checkmarx SCA - Scan Summary & Details**
### Cx-SCA Summary
**Total Packages Identified**: **98**
**Scan Risk Score**: **9.80**

High&nbsp;**2** **High** **severity vulnerabilities**
Medium&nbsp;**0** **Medium** **severity vulnerabilities**
Low&nbsp;**0** **Low** **severity vulnerabilities**
[View more details on Checkmarx UI](https://eu.sca.checkmarx.net/#/projects/4782c9b4-729a-468b-b8b0-46bb71cb54e7/reports/3ce1ea2e-c545-4c39-adc2-1b51fc5dc67c)
### Cx-SCA vulnerability result overview
|Vulnerability ID|Package|Severity|CVSS score|Publish date|Current version|Recommended version|Link in CxSCA|Reference – NVD link|
---|---|---|---|---|---|---|---|---
`CVE-2016-1000027`|org.springframework:spring-web|HIGH|9.8|2020-01-02T23:15:00|5.3.2|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/4782c9b4-729a-468b-b8b0-46bb71cb54e7/reports/3ce1ea2e-c545-4c39-adc2-1b51fc5dc67c/vulnerabilities/CVE-2016-1000027%3AYarn-org.springframework%3Aspring-web-5.3.2/vulnerabilityDetails)|[CVE-2016-1000027](https://nvd.nist.gov/vuln/detail/CVE-2016-1000027)
`CVE-2016-1000027`|org.springframework:spring-webmvc|HIGH|9.8|2020-01-02T23:15:00|5.3.2|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/4782c9b4-729a-468b-b8b0-46bb71cb54e7/reports/3ce1ea2e-c545-4c39-adc2-1b51fc5dc67c/vulnerabilities/CVE-2016-1000027%3AYarn-org.springframework%3Aspring-webmvc-5.3.2/vulnerabilityDetails)|[CVE-2016-1000027](https://nvd.nist.gov/vuln/detail/CVE-2016-1000027)

**Review by natalieb07 — APPROVED** [review]

File: src/main/java/com/checkmarx/controller/WebController.java [hunk 1]
```diff
@@ -2,36 +2,33 @@
...
-    ApplicationContext applicationContext;
+    private final ApplicationContext applicationContext;
     
-    @Autowired
-    ConfigurationService genericScmService;
+    private final ConfigurationService genericScmService;
 
     //no special character regex validation
-    private static final String VALIDATION_REGEX = "^[^`~!@#$%^&*+={}:;<>?๐฿]*$";
+    private static final String VALIDATION_REGEX = "^[^`~!@#$%^&*+={}:;<>?]*$";
```
[the remark this answers is not in the document]
Comment (alex-ko-dev) [reply src/main/java/com/checkmarx/controller/WebController.java:31]: I noticed 2 strange characters in the regex. Asaf confirmed they were in the regex by mistake. So I've removed them.

File: src/main/java/com/checkmarx/service/AbstractScmService.java [hunk 2]
```diff
@@ -2,57 +2,62 @@
...
-     * @param accessToken access token generated before using GitHub api, Gives access to relevant
-     *                  GitHub data
-     * @return true if verification passed successfully
+     * Only checks if the access token is not empty. Additional token validation depends on a specific SCM.
      */
-    protected boolean verifyAccessToken(AccessTokenDto accessToken) {
-        return accessToken != null && accessToken.getAccessToken() != null && !accessToken.getAccessToken().isEmpty();
+    protected void verifyAccessToken(AccessTokenDtoBase tokenDto) {
+        if (tokenDto == null || StringUtils.isEmpty(tokenDto.getAccessToken())) {
+            log.error(RestWrapper.GENERATE_ACCESS_TOKEN_FAILURE);
```
[the remark this answers is not in the document]
Comment (alex-ko-dev) [reply src/main/java/com/checkmarx/service/AbstractScmService.java]: Fixed.

**Pull request merged**